manual/src/cmds/native-debugger.etex

```diff
@@ -0,0 +1,389 @@
+\chapter{Native debugger} \label{c:native-debugger}
+%HEVEA\cutname{native-debugger.html}
+
+\section{s:native-debugger-preliminaries}{Preliminaries}
+
+This chapter describes the support for debugging OCaml executables built with the native-code compiler \texttt{ocamlopt}, using GDB or LLDB on the Linux, macOS, and FreeBSD platforms. We will call this \texttt{native debugging}, in contrast to bytecode debugging supported via \texttt{ocamldebug} (see chapter~\ref{c:debugger}).
+
+Presently there is no Windows support for native debugging of OCaml.
+
+\subsection{ss:native-debugger-dwarf}{DWARF}
+
+The OCaml compiler uses the \href{http://dwarfstd.org/}{DWARF} debugging format to describe the debug information it generates.  DWARF is a debugging information format used by many compilers and debuggers to support source-level debugging. It has support for Linux ELF, macOS Mach-O and FreeBSD ELF executable formats.
+
+Within the larger DWARF standard, the compiler uses Call Frame Information (CFI) to describe the stack frames for OCaml code, which allows for unwinding the call stack to generate a backtrace. CFI is preserved across OCaml stack frames into sections of the runtime written in C, and across the Foreign Function Interface (FFI) if the foreign language also provides CFI information. OCaml defines its own calling convention,which details how arguments are passed to functions, how values are returned from functions, and how registers are used. This information is also encoded using CFI.
+
+The OCaml compiler generates line information that maps machine instructions back to their source location (e.g., the instruction at address \texttt{0xdeadbeef} originated from \texttt{myprogram.ml} line 42). This allows native debuggers to display the OCaml source code for the program being debugged and enables stepping through OCaml source code.
+
+\subsection{ss:native-debugger-name-mangling}{Name Mangling}
+
+Name mangling describes how the OCaml compiler generates symbol names for OCaml language constructs. The format of these symbols is important for native debuggers to uniquely identify the source function for a symbol without referencing the original source code. In the absence of source mappings you would use mangled names to set breakpoints, and they appear in debugger output like backtraces. Understanding OCaml's name mangling is therefore useful when debugging OCaml programs.
+
+OCaml 5.4 onwards uses a name mangling scheme of \texttt{caml<MODULE_NAME>\$<FUNCTION_NAME>_<NNN>}, where \texttt{NNN} is a randomly generated number.
+
+Note: OCaml 5.3 uses a dot \texttt{.} as the separator. For the Windows MSVC port (restored in OCaml 5.3), the scheme uses \texttt{\$} as the separator. Before OCaml 5.1.1 the scheme used two underscores as the separator, e.g., \texttt{caml<MODULE_NAME>__<FUNCTION_NAME>_<NNN>}.
+
```

Comment:
Unfortunately, it seems that this section will have to be updated depeding on the fate of https://github.com/ocaml/ocaml/issues/14143 .